MaxText/pyconfig.py

```diff
@@ -94,6 +95,11 @@ def _load_kwargs(self, argv: list[str], **kwargs):
   @staticmethod
   def user_init(raw_keys):
     '''Transformations between the config data and configs used at runtime'''
+
+    # We initialize the jax distributed system here because it must be done before device backend is initialized.
+    if raw_keys["enable_checkpointing"] and raw_keys["async_checkpointing"] and raw_keys["compile_topology_num_slices"]==-1:
```

Comment:
Why can't we always do it here?

Reply:
We will work on making it so that it can always be done - but currently arguments only automated for cloud TPU environments - not CPU. To allow fast tests and development on CPU or GPU environments we can skip the jax distributed system by disabling async checkpointing - but in the future we should create initializers for CPU and GPU environments